controllers/configurationpolicy_controller.go

```diff
@@ -2725,6 +2724,16 @@ func (r *ConfigurationPolicyReconciler) checkAndUpdateResource(
 				return false, "", false, false
 			}
 
+			// Generate and log the diff
+			if objectT.RecordDiff == policyv1.RecordDiffLog {
```

Comment:
By having this here, I guess inform on OCP 3.11 will not log the diff since DryRun is not supported on OCP 3.11.

Reply:
That sounds like acceptable risk to me?

Reply:
...or should make a best effort to log something?

Reply:
So in OCP 3.11, we return right after the first mismatch if it's inform:
https://github.com/open-cluster-management-io/config-policy-controller/blob/77048df3ca61b976d61f75ec9b1b93b48400c168/controllers/configurationpolicy_controller.go#L2638-L2644

I'm fine with not supporting OCP 3.11 if we document that. If you want to support it, you basically need to remove the linked code and then add `else if isInform` to the `if r.DryRunSupported {` statement and log the diff and return.

Reply:
> ...and log the diff and return.

But what's the diff? The diff between the `objectDefinition` and the object on the cluster? Seems there'd need to be some extra logic there to only diff the fields that are defined if it's `musthave`, right?

Reply:
I think `checkAndUpdateResource` is only called if it's musthave or mustonlyhave and the object exists.